src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -190,30 +194,31 @@ private ImmutableArray<BoundPattern> BindSubPropertyPatterns(PropertyPatternSynt
                         diagnostics.Add(node, useSiteDiagnostics);
                     }
 
-                    properties.Add(property);
-                    pattern = this.BindPattern(syntax.Pattern, null, property.GetTypeOrReturnType(), hasErrors, diagnostics);
+                    var propertyType = property.GetTypeOrReturnType();
+                    pattern = this.BindPattern(syntax.Pattern, null, propertyType, hasErrors, diagnostics);
+                    result.Add(new BoundSubPropertyPattern(syntax, propertyType, property, resultKind, pattern, hasErrors));
                 }
                 else
                 {
                     Error(diagnostics, ErrorCode.ERR_NoSuchMember, propName, type, propName.Identifier.ValueText);
                     pattern = new BoundWildcardPattern(node, hasErrors: true);
+                    result.Add(new BoundSubPropertyPattern(syntax, CreateErrorType(), null, resultKind, pattern, true));
```

Comment:
I agree that it would be valuable to review this design issue in the context of the binder as a whole. I do not want to start that review by making a change in this PR that is inconsistent with the existing design.